xtask/src/ast_src.rs

```diff
@@ -607,55 +1972,165 @@ pub(crate) const AST_SRC: AstSrc = AstSrc {
             const_args: [ConstArg],
             T![>]
         }
+
+        /// Type argument that is passed at generic instantiation site.
+        ///
+        /// ```
+        /// use foo::<'a, ❰ u64 ❱, ❰ bool ❱, Item = Bar, 42>::baz;
+        /// ```
+        ///
+        /// [Reference](https://doc.rust-lang.org/reference/paths.html#paths-in-expressions)
         struct TypeArg { TypeRef }
+
+        // TODO: verify inline type bounds example
+        /// Associated type argument that is passed at generic instantiation site.
+        /// ```
+        /// use foo::<'a, u64, bool, ❰ Item = Bar ❱, 42>::baz;
+        ///
+        /// trait Bruh<T>: Iterator<❰ Item: Debug ❱> {}
+        /// ```
+        ///
         struct AssocTypeArg : TypeBoundsOwner { NameRef, T![=], TypeRef }
+
+        // TODO: verify?
+        /// Lifetime argument that is passed at generic instantiation site.
+        ///
+        /// ```
+        /// fn foo<'a>(s: &'a str) {
+        ///     bar::<❰ 'a ❱>(s);
+        /// }
+        /// ```
+        ///
+        /// [Reference](https://doc.rust-lang.org/reference/paths.html#paths-in-expressions)
         struct LifetimeArg { T![lifetime] }
+
+        // TODO: does this peratain to const generics?
+        // What does equal sign do here?
+        /// Constant value argument that is passed at generic instantiation site.
+        ///
+        /// ```
+        /// foo::<❰ u32 ❱, ❰ true ❱ >();
+        ///
+        /// bar::<❰ { 2 + 2} ❱>();
+        /// ```
+        ///
+        /// [RFC](https://github.com/rust-lang/rfcs/blob/master/text/2000-const-generics.md#declaring-a-const-parameter)
         struct ConstArg { Literal, T![=], BlockExpr }
 
-        struct MacroItems: ModuleItemOwner{ }
+        // TODO: Idk what I am writing here, please don't believe these words.
+        // TODO: clarify @matklad @edwin0cheng
+        /// Macro items is a node that holds all the items created by expanding a macro.
+        ///
+        /// ```
+        /// foo!(); // expands into some items -v
+        ///         // ❰ struct Foo; impl Bar for Foo; ❱
+        /// ```
+        ///
+        /// [Reference](https://doc.rust-lang.org/reference/macros.html)
+        struct MacroItems: ModuleItemOwner { }
 
+        // TODO: Idk what I am writing here, please don't believe these words.
+        // TODO: clarify @matklad @edwin0cheng
```

Comment:
It is not in used right now, I think it could be removed.

Reply:
@edwin0cheng removed this in 7a49165f5d5c8186edd04f874eae8a98e39d3df6

Reply:
Having removed this node I get the following warning when generating the nodes:
```
Warning: node MacroStmts not defined in ast source
```
This is because there is an appropriate syntax node kind which is also utilized by
[ra_parser](https://github.com/rust-analyzer/rust-analyzer/blob/b30fb9e0998918390fb614726b5ed3ee65e29a35/crates/ra_parser/src/grammar.rs#L136)

I'll revert this commit and return it back. I don't know which particular doc comment it requires, I'll put a `FIXME(@edwin0cheng)` there

Reply:
Okay, I will remove it later.